includes/data-events/listeners.php

```diff
@@ -100,3 +100,104 @@ function( $provider, $email, $lists_added, $lists_removed, $result ) {
 		];
 	}
 );
+
+/**
+ * For when there's a new donation through WooCommerce.
+ */
+Data_Events::register_listener(
+	'newspack_donation_order_processed',
+	'donation_new',
+	function( $order_id, $product_id ) {
+		$order = \wc_get_order( $order_id );
+		if ( ! $order ) {
+			return;
+		}
+		return [
+			'user_id'       => $order->get_customer_id(),
+			'email'         => $order->get_billing_email(),
+			'amount'        => (float) $order->get_total(),
+			'currency'      => $order->get_currency(),
+			'recurrence'    => get_post_meta( $product_id, '_subscription_period', true ),
+			'platform'      => 'newspack',
+			'platform_data' => [
+				'order_id'   => $order_id,
+				'product_id' => $product_id,
+			],
+		];
+	}
+);
+
+/**
+ * For when there's a new donation through the Stripe platform.
+ */
+Data_Events::register_listener(
+	'newspack_new_donation_woocommerce',
+	'donation_new',
+	function( $order, $client_id ) {
+		$order_id = $order->get_id();
+		return [
+			'user_id'       => $order->get_customer_id(),
+			'email'         => $order->get_billing_email(),
+			'amount'        => (float) $order->get_total(),
+			'currency'      => $order->get_currency(),
+			'recurrence'    => get_post_meta( $product_id, '_subscription_period', true ),
+			'platform'      => 'stripe',
+			'platform_data' => [
+				'order_id'   => $order_id,
+				'product_id' => \Newspack\Donations::get_order_donation_product_id( $order_id ),
+				'client_id'  => $client_id,
+			],
+		];
+	}
+);
+
+/**
+ * For when there's a new donation subscription.
+ *
+ * This will be fetched from a new donation, so we're hooking into the 'donation_new' dispatch.
+ */
+Data_Events::register_listener(
+	'newspack_data_event_dispatch_donation_new',
+	'donation_subscription_new',
```

Comment:
Testing these steps:

> 3. Make a new one-time donation using the donation block
> 4. Confirm you see the logs with [NEWSPACK-DATA-EVENTS][Newspack\Data_Events::dispatch]: Dispatching action "donation_new" via Action Scheduler.

Making a one-time donation, I see both `donation_subscription_new` and `donation_new` actions being dispatched.

```
[12-Jan-2023 23:31:07 UTC] [NEWSPACK-DATA-EVENTS][Newspack\Data_Events::dispatch]: Dispatching action "donation_subscription_new" via Action Scheduler.
[12-Jan-2023 23:31:07 UTC] [NEWSPACK-DATA-EVENTS][Newspack\Data_Events::dispatch]: Dispatching action "donation_new" via Action Scheduler.
```

I also see this in the Scheduled Actions log:

<img width="1806" alt="Screen Shot 2023-01-12 at 5 17 26 PM" src="https://user-images.githubusercontent.com/2230142/212208157-68673969-79d2-4607-b997-d69a0c8e79a3.png">

Note also the empty timestamps of the scheduled actions—not sure if that's a bug or not?

Reply:
Can confirm that the `donation_subscription_new` is only fired on new subscriptions now, not one-time donations.

The timestamps still look like `0000-00-00`, but I'm not sure if this is a bug.